app/domain/authentication/authn_k8s/web_socket_client_event_handler.rb

```diff
@@ -0,0 +1,109 @@
+# frozen_string_literal: true
+
+module Authentication
+  module AuthnK8s
+
+    class WebSocketClientEventHandler
+
+      attr_reader :channel_closed, :message_log
+
+      def initialize(
+        ws_client:,
+        stdin:,
+        body:,
+        pod_name:,
+        validate_message:,
+        message_log_class:,
+        logger:
+      )
+        @ws_client = ws_client
+        @stdin = stdin
+        @body = body
+        @pod_name = pod_name
+        @validate_message = validate_message
+        @message_log_class = message_log_class
+        @logger = logger
+
+        @message_log = @message_log_class.new
+        @channel_closed = false
+      end
+
+      def on_open
+        hs       = @ws_client.handshake
+        hs_error = hs.error
+
+        if hs_error
+          # Add the error to the stream instead of raising it so we continue to
+          # receive other messages. At the end of this class run we will verify
+          # the stream to see if any errors were written
```

Comment:
I would think an error during the handshake is a fatal error.  What's the reason it's not?

Reply:
what do you mean by fatal? from what i know, fatal error should cause the server to crash which is not the case here. In the current implementation, the authentication request will responds with 401. Is this ok?

Reply:
I meant that if we can't complete the handshake, isn't the whole prospect of communication dead?

Reply:
It's fatal to the this session and to the client.